types/underscore/index.d.ts

```diff
@@ -154,6 +154,10 @@ declare module _ {
         : TItem
         : T;
 
+    type AnyFalsy = undefined | null | false | '' | 0;
```

Comment:
A quick note here: `-0` is treated as `0` by TS, `NaN` is not a constant with its own dedicated constant type at this time, and `0n` can't be added until TS3.2 is the minimum TS version without adding a versioned type definition.

Reply:
I had generally thought that `void` was more or less an alias for `undefined`, but it looks like it might in reality be less since `void extends undefined` is false.

Actually, though, `undefined extends void` is true, so I guess they are related a little. I should ideally use `void` instead of `undefined` here though to cover `void` as well.

Reply:
Oh, the linter doesn't like `void` being used outside of the context of a return type so making this change will fail the CI, oh well I guess.